Tasks/AzureAppServiceManage/azureappservicemanage.ts

```diff
@@ -0,0 +1,96 @@
+        tl.warning(exception);
+    }
+}
+
+async function run() {
+    try {
+        tl.setResourcePath(path.join( __dirname, 'task.json'));
+        var connectedServiceName = tl.getInput('ConnectedServiceName', true);
+        var action = tl.getInput('Action', true);
+        var webAppName: string = tl.getInput('WebAppName', true);
+        var resourceGroupName: string = tl.getInput('ResourceGroupName', false);
+        var sourceSlot: string = tl.getInput('SourceSlot', false);
+        var swapWithProduction = tl.getBoolInput('SwapWithProduction', false);
+        var targetSlot: string = tl.getInput('TargetSlot', false);
+        var preserveVnet: boolean = tl.getBoolInput('PreserveVnet', false);
+        var endPointAuthCreds = tl.getEndpointAuthorization(connectedServiceName, true);
+        var subscriptionId = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true);
+        var taskResult = true;
+
+        var endPoint = new Array();
+        endPoint["servicePrincipalClientID"] = tl.getEndpointAuthorizationParameter(connectedServiceName, 'serviceprincipalid', true);
+        endPoint["servicePrincipalKey"] = tl.getEndpointAuthorizationParameter(connectedServiceName, 'serviceprincipalkey', true);
+        endPoint["tenantID"] = tl.getEndpointAuthorizationParameter(connectedServiceName, 'tenantid', true);
+        endPoint["subscriptionId"] = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true);
+        endPoint["url"] = tl.getEndpointUrl(connectedServiceName, true);
+
+        if(resourceGroupName === null) {
+            resourceGroupName = await azureRmUtil.getResourceGroupName(endPoint, webAppName);
+        }
+        switch(action) {
+            case "Start Azure App Service": {
+                tl._writeLine(tl.loc('StartingAppService', webAppName));
+                tl._writeLine(await azureRmUtil.startAppService(endPoint, resourceGroupName, webAppName));
+                break;
+            }
+            case "Stop Azure App Service": {
+                tl._writeLine(tl.loc('StoppingAppService', webAppName));
+                tl._writeLine(await azureRmUtil.stopAppService(endPoint, resourceGroupName, webAppName));
+                break;
+            }
+            case "Restart Azure App Service": {
+                tl._writeLine(tl.loc('RestartingAppService', webAppName));
+                tl._writeLine(await azureRmUtil.restartAppService(endPoint, resourceGroupName, webAppName));
+                break;
+            }
+            case "Swap Slots": {
+                await swapSlot(endPoint, resourceGroupName, webAppName, sourceSlot, swapWithProduction, targetSlot, preserveVnet);
+                break;
+            }
+            default:
+                throw Error(tl.loc('InvalidAction'));
+        }
+    }
+    catch(exception)
+    {
+        taskResult = false;
+        tl.setResult(tl.TaskResult.Failed, exception);
+    }
+    var customMessage = {
+        type: action
```

Comment:
type should be "SlotSwap" (for swap slot action) since we are using this hard coded value to check for type of log in Ibiza CD flow.